docs/tests.md

```diff
@@ -0,0 +1,439 @@
+This test checks operator behaviour to manage and maintain MAPI components deployment state.
+
+#### Machine API operator deployment should reconcile mutating webhook configuration
+#### Machine API operator deployment should reconcile validating webhook configuration
+* Parallel: No (each)
+* ExecTime: ~5s on AWS (each)
+* Recommendation:
+  - Extract this test, make it parallel
+  - break dependency to the vendored MAPO if it's possible
+  - this two tests might be generalized in a way and merged into one
+
+Two pretty much the same tests. Checks that webhooks are configured in a cluster. Tests using webhook definition (golang structure) imported from the MAPO
+repository, which is not good and might cause issues if webhook settings will need to be changed (these tests will fail, till MAPO revendoring in cluster-actuator-pkg-repo).
+
+This test checks operator behaviour to manage and maintain MAPI webhooks state.
+
+#### Machine API operator deployment should recover after validating webhook configuration deletion
+#### Machine API operator deployment should recover after mutating webhook configuration deletion
+* Parallel: No (each)
+* ExecTime: ~5s on AWS (each)
+* Recommendation:
+    - break dependency to the vendored MAPO if it's possible
+    - this two tests might be generalized in a way and merged into one, the only difference is a resource type
+
+This test checks that webhook configurations are presented in cluster then delete it. Waits for configurations to be restored and 
+being equal to what is encoded within the vendored MAPO repo.
+
+This test checks operator behaviour to manage and maintain MAPI webhooks state.
+
+#### Machine API operator deployment should maintains spec after mutating webhook configuration change and preserve caBundle
+#### Machine API operator deployment should maintains spec after validating webhook configuration change and preserve caBundle
+* Parallel: No (each)
+* ExecTime: ~5s on AWS (each)
+* Recommendation:
+    - break dependency to the vendored MAPO if it's possible
+    - this two tests might be generalized in a way and merged into one, the only difference is a resource type
+
+This test checks that webhook configurations are presented in cluster then changes it. Waits for configurations to be restored and
+being equal to what is encoded within the vendored MAPO repo.
+
+This test checks operator behaviour to manage and maintain MAPI webhooks state.
+
+#### Machine API cluster operator status should be available
+* Parallel: Yes
+* ExecTime: ~2s on AWS
+
+Checks cluster operator status to be available, not progressing and not degraded.
+
+#### When cluster-wide proxy is configured, Machine API cluster operator should  create machines when configured behind a proxy
+* Parallel: No
+* ExecTime: ~30m on AWS
+* Recommendation:
+  - ~make this test periodic (done already)~
+  - extend this test - add check for the CCCMO also
```

Comment:
I don't think so, since cluster autoscaler deals with machine resource through the kubernetes API.

Machine controllers and CCMs interact with cloud APIs, and setting up proxy properly is crucial here.
IIRC we have had bugs for CCMs and proxy issues.